numpy/core/tests/test_multiarray.py

```diff
@@ -7013,6 +7013,37 @@ def test_null_inside_ustring_array_is_truthy(self):
         assert_(a)
 
 
+class TestFormat(object):
+
+    def test_0d(self):
+        a = np.array(np.pi)
+        assert_equal('{:0.3g}'.format(a), '3.14')
+        assert_equal('{:0.3g}'.format(a[()]), '3.14')
+
+    def test_1d_no_format(self):
+        a = np.array([np.pi])
+        assert_equal('{}'.format(a), str(a))
+
+    def test_1d_format(self):
+        # until gh-5543, ensure that the behaviour matches what it used to be
+        a = np.array([np.pi])
+
+        def ret_and_exc(f, *args, **kwargs):
+            try:
+                return f(*args, **kwargs), None
+            except Exception as e:
+                # exceptions don't compare equal, so return type and args
+                # which do
+                return None, (type(e), e.args)
+
+        # Could switch on python version here, but all we care about is
+        # that the behaviour hasn't changed
+        assert_equal(
+            ret_and_exc(object.__format__, a, '30'),
```

Comment:
I suppose we should silence warnings in `ret_and_exc`, or return a third tuple item which is the warning

Reply:
Maybe just figure out what the result should be and use that? Eventually `object.__format__` is going to fail with non-empty argument.

Reply:
Right, and I think that `ndarray.__format__` should start failing at the same time. Maybe we should just make `ndarray.__format__` fail with an empty argument right now...

Reply:
Looks like it went away in Python 3.4.
```
* :meth:`object.__format__` no longer accepts non-empty format strings, it now
  raises a :exc:`TypeError` instead.  Using a non-empty string has been
  deprecated since Python 3.2.  This change has been made to prevent a
  situation where previously working (but incorrect) code would start failing
  if an object gained a __format__ method, which means that your code may now
  raise a :exc:`TypeError` if you are using an ``'s'`` format code with objects
  that do not have a __format__ method that handles it.  See :issue:`7994` for
  background.
```
Should ours work if an empty string is passed?

Reply:
I fixed this by making the test Python version dependent.